### DISCUSSION TITLE
/history 200 response always empty array

### DISCUSSION THREAD

**Discussion by 17jmumford** [amzn/ads-advanced-tools-docs#88] [Q&A]
Howdy! I am requesting data from the /history [endpoint](https://advertising.amazon.com/API/docs/en-us/change-history/#/default/getHistory). I am getting an empty response back. Here is an example of my request and response:

payload: "{"fromDate":1678860000000,"toDate":1684330400000,"eventTypes":{"KEYWORD":{}}}", 

resp: {:events=>[], :page_size=>50, :page_offset=>0, :max_page_number=>0, :total_records=>0}

I've tried all the historyEventTypes with windows ranging from two days to two months. No luck.  Has anyone had luck with this endpoint? Any suggestions would be much appreciated!

**Comment by SylvieManx** [comment]
I found mine too, no data returned

**Comment by elizafarley** [comment]
Hi @17jmumford, were you able to find a solution to this issue?

I know right now we don't have any supporting content for this endpoint, but would you (or others) find a tutorial with code samples helpful?

**Comment by stevieamazon21** [comment]
hey @17jmumford ,

Looks to me like your response is empty because your payload is too broad.  To provide more details, you aren't including any eventTypes or filters for the keyword.  Here's the relevant portion of our documentation, https://advertising.amazon.com/API/docs/en-us/change-history/#/default/getHistory:
![image-20230523-011935](https://github.com/amzn/ads-advanced-tools-docs/assets/105813247/82d6c092-d93d-4f3c-98b7-49713cd5b310)
Can you try defining this portion of the object, and tell me whether the empty response persists?

Kindly,
Steve Berliner | API Developer Support Engineer | Amazon
